From 1500f06aee94dd2d83a0d59b83705f1221692e2e Mon Sep 17 00:00:00 2001
From: Jeff Layton <jlayton@redhat.com>
Date: Thu, 17 Apr 2014 19:10:03 -0400
Subject: [security] have cap_dentry_init_security return error

Message-id: <1397761803-19540-1-git-send-email-jlayton@redhat.com>
Patchwork-id: 78964
O-Subject: [RHEL7.1 PATCH] [BZ#1070910] security: have cap_dentry_init_security return error
Bugzilla: 1070910
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

Currently, cap_dentry_init_security returns 0 without actually
initializing the security label. This confuses its only caller
(nfs4_label_init_security) which expects an error in that situation, and
causes it to end up sending out junk onto the wire instead of simply
suppressing the label in the attributes sent.

When CONFIG_SECURITY is disabled, security_dentry_init_security returns
-EOPNOTSUPP. Have cap_dentry_init_security do the same.

Signed-off-by: Jeff Layton <jlayton@redhat.com>
Acked-by: Serge E. Hallyn <serge.hallyn@ubuntu.com>
Signed-off-by: James Morris <james.l.morris@oracle.com>
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1070910
Upstream: d4a141c8e77043bd674dd6aa0b40bc3675cb7b1d
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/security/capability.c b/security/capability.c
index 32b5157..7817f7a 100644
--- a/security/capability.c
+++ b/security/capability.c
@@ -116,7 +116,7 @@ static int cap_dentry_init_security(struct dentry *dentry, int mode,
      struct qstr *name, void **ctx,
      u32 *ctxlen)
 {
- return 0;
+ return -EOPNOTSUPP;
 }
 
 static int cap_inode_alloc_security(struct inode *inode)
-- 
1.7.1